platform/javascript/api/javascript_tools_editor_plugin.cpp

```diff
@@ -58,23 +59,40 @@ JavaScriptToolsEditorPlugin::JavaScriptToolsEditorPlugin(EditorNode *p_editor) {
 
 void JavaScriptToolsEditorPlugin::_download_zip(Variant p_v) {
 	if (!Engine::get_singleton() || !Engine::get_singleton()->is_editor_hint()) {
-		WARN_PRINT("Project download is only available in Editor mode");
+		ERR_PRINT("Downloading the project as a ZIP archive is only available in Editor mode.");
 		return;
 	}
 	String resource_path = ProjectSettings::get_singleton()->get_resource_path();
 
 	FileAccess *src_f;
 	zlib_filefunc_def io = zipio_create_io_from_file(&src_f);
-	zipFile zip = zipOpen2("/tmp/project.zip", APPEND_STATUS_CREATE, nullptr, &io);
-	String base_path = resource_path.substr(0, resource_path.rfind("/")) + "/";
+
+	// Name the downloded ZIP file to contain the project name and download date for easier organization.
+	// Replace characters not allowed (or risky) in Windows file names with safe characters.
+	// In the project name, all invalid characters become an empty string so that a name
+	// like "Platformer 2: Godette's Revenge" becomes "platformer_2-_godette-s_revenge".
+	const String project_name_safe =
+			GLOBAL_GET("application/config/name").to_lower().replace(" ", "_");
+	const String datetime_safe =
+			Time::get_singleton()->get_datetime_string_from_system(false, true).replace(" ", "_");
+	const String output_name = OS::get_singleton()->get_safe_dir_name(vformat("%s_%s.zip"));
+	const String output_path = String("/tmp").plus_file(output_name);
```

Comment:
How does this work on Windows?

We should use Godot's own cache folder for that as we do for various other tmp objects (see e.g. Android export code with `tmp_unaligned.apk`).

Reply:
Well, the `/tmp` is the folder in the emscripten fake FS, and that's always unix like.

Reply:
Oh ok, makes sense.

There might be a system function to get a valid tmp file which could be used instead of making one ourselves though. (Like `mktemp` on Unix.)

But this should be fine as is too.